src/Model/HasLegacyAttributesTrait.php

```diff
@@ -122,6 +138,10 @@ public function cleanAttribute(string $key)
      */
     public function setAttribute(string $key, $value)
     {
+        if ($caster = CastResolver::resolve($this->casts[$key] ?? null)) {
```

Comment:
The same solution described before can be used here.

Reply:
Done: https://github.com/leroy-merlin-br/mongolid/pull/220/commits/0fa4607f1c7a1c45c9c09bd8411843350ac6dbe7